### PR TITLE
gnome-decoder: init at 0.3.1

### PR DESCRIPTION
**Pull request by zendo**
###### Description of changes

 #162982 

Scan and Generate QR Codes
https://gitlab.gnome.org/World/decoder

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by bjornfor** [comment]
Looks good to me, but would like input from @jtojnar about where the expression should be located. (I see there are several GNOME apps in pkgs/desktops/gnome/apps/, but also some outside that directory. Is there a system here?)

**Comment by jtojnar** [comment]
We are trying to disband `pkgs/desktops/gnome/apps` so `pkgs/applications` look good to me.

**Review by bjornfor — APPROVED** [review]

File: pkgs/applications/graphics/gnome-decoder/default.nix [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+in
+stdenv.mkDerivation rec {
+  pname = "gnome-decoder";
+  version = "0.3.1";
+
+  src = fetchFromGitLab {
+    domain = "gitlab.gnome.org";
+    owner = "World";
+    repo = "decoder";
+    rev = "${version}";
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/gnome-decoder/default.nix:43]:
```suggestion
    rev = version;
```

File: pkgs/applications/graphics/gnome-decoder/default.nix [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  };
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit src;
+    name = "${pname}-${version}";
+    sha256 = "sha256-RMHVrv/0q42qFUXyd5BSymzx+BxiyqTX0Jzmxnlhyr4=";
+  };
+
+  nativeBuildInputs = [
+    clang
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/gnome-decoder/default.nix:54]: Do we want to use the clangStdenv?

File: pkgs/applications/graphics/gnome-decoder/default.nix [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    glib
+    gtk4
+    libadwaita-git
+    zbar
+    sqlite
+    pipewire
+  ] ++ (with gst_all_1; [
+    gstreamer
+    gst-plugins-base
+    (gst-plugins-bad.override { enableZbar = true; })
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/gnome-decoder/default.nix:77]: This should be moved to top-level.

File: pkgs/applications/graphics/gnome-decoder/default.nix [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  libadwaita-git = libadwaita.overrideAttrs (oldAttrs: rec {
+    version = "1.2.alpha";
+    src = fetchFromGitLab {
+      domain = "gitlab.gnome.org";
+      owner = "GNOME";
+      repo = "libadwaita";
+      rev = version;
+      hash = "sha256-JMxUeIOUPp9k5pImQqWLVkQ2GHaKvopvg6ol9pvA+Bk=";
+    };
+  });
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/gnome-decoder/default.nix:33]: We should move this to top-level.

File: pkgs/applications/graphics/gnome-decoder/default.nix (hunk 2 above)
Comment (jtojnar) [reply pkgs/applications/graphics/gnome-decoder/default.nix:54]: This is probably just needed for some Rust package. A comment might be nice.

File: pkgs/applications/graphics/gnome-decoder/default.nix (hunk 3 above)
Comment (zendo) [reply pkgs/applications/graphics/gnome-decoder/default.nix:77]: I don't know what your meaning.

File: pkgs/top-level/all-packages.nix [hunk 5]
```diff
@@ -6696,6 +6696,8 @@ with pkgs;
 
   gnome-desktop = callPackage ../development/libraries/gnome-desktop { };
 
+  gnome-decoder = callPackage ../applications/graphics/gnome-decoder { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:6699]:
```suggestion
  gnome-decoder = callPackage ../applications/graphics/gnome-decoder {
    inherit (gst_all_1) gstreamer gst-plugins-base;
    gst-plugins-bad = gst_all_1.gst-plugins-bad.override { enableZbar = true; }
  };
```

File: pkgs/applications/graphics/gnome-decoder/default.nix [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+, meson
+, ninja
+, pkg-config
+, glib
+, gtk4
+, libadwaita
+, zbar
+, sqlite
+, pipewire
+, gst_all_1
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/gnome-decoder/default.nix:16]:
```suggestion
, gstreamer
, gst-plugins-base
, gst-plugins-bad
```

**Pull request closed without merging**